lua/telescope/_extensions/file_browser/actions.lua

```diff
@@ -417,6 +417,37 @@ fb_actions.goto_cwd = function(prompt_bufnr)
   current_picker:refresh(finder, { reset_prompt = true, multi = current_picker._multi })
 end
 
+--- Change working directory of nvim to the selected file/folder in |builtin.file_browser|.
+---@param prompt_bufnr number: The prompt bufnr
+fb_actions.change_cwd = function(prompt_bufnr)
+  local picker = action_state.get_current_picker(prompt_bufnr)
+  local finder = picker.finder
+  local sel = fb_utils.get_selected_files(prompt_bufnr, true)[1]
+  finder.path = sel:is_dir() and sel:absolute() or sel:parent():absolute()
+  finder.cwd = finder.path
+  finder.files = true
+  vim.cmd("cd " .. finder.path)
+  if picker.results_border then
+    local title = Path:new(finder.path):make_relative(finder.path) .. os_sep
+    picker.results_border:change_title(title)
+  end
+  picker:refresh(finder, { reset_prompt = true, multi = picker._multi })
+end
```

Comment:
I think we should `vim.api.nvim_echo` or `print` a success message here, especially if we don't otherwise give visual feedback (which are not universally handled atm).